### PR TITLE
Improved Docker setup and documentation

### PR DESCRIPTION
**Pull request by metachris**
* created `/docker` directory with all Docker stuff inside
* 2 Dockerfiles now, one for dev and one for master
* docker-compose file to setup neoscan with a private network
* Documentation

**How did you make sure your solution works?**

Manual tests

**Did you add any tests?**

No

**Are there any special changes in the code that we should be aware of?**

No

**Did you run `make lint` and `make test`?**

Yes

**Are you making a PR to a feature branch or development rather than master?**

dev

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# to a private network, make sure the privatenet container is already running.
+# https://hub.docker.com/r/cityofzion/neo-privatenet
+#
+# Start a container interactively, opening a bash in `/neo-python`, and mounting the current directory as `/neo-python/sc`:
+#
+#    $ docker run --rm -it -v $(pwd):/neo-python/sc --net=host -h neopython --name neopython neopython /bin/bash
+#
+# Once you are inside the container, you can start neo-python with `python3 prompt.py -p` (using -p to connect to a private net).
+# To update neo-python, just run `git pull` and `pip install -e .`
+FROM ubuntu:16.04
```
Comment (revett) [inline docker/Dockerfile]: Why not use `python:3.5` as the base image? 

Benefits:

- Smaller image size
- No need to install Python dependencies
- Faster image build time

Comment (metachris) [reply docker/Dockerfile]: The reason is that the Python image need more space! I did test the builds for both `python:3.5` and `ubuntu:16.04` and the Python image uses 200MB more compared to Ubuntu.

**Comment by revett** [comment]
@metachris Is it worth having a README within the `docker/` directory instead of adding documentation to the files themselves.

To me it hides the documentation by having them in the file - SHOW OFF the documentation 😊

File: docker/Dockerfile (hunk 1 above)
Comment (revett) [reply docker/Dockerfile]: Ignore me, I thought it was the other way round 😂

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+# Install dependencies
+RUN apt-get update && apt-get install -y \
+    wget \
+    git-core \
+    python3-dev \
```
Comment (revett) [inline docker/Dockerfile]: @metachris Do we need to define this as Python 3.5?

Comment (metachris) [reply docker/Dockerfile]: yes

**Comment by metachris** [comment]
Yes, I think having a README is worth it -- it's context for people that click into the `docker` folder on GitHub, and not much effort to maintain. Makes it overall just a little easier for new people.

But perhaps we can make it shorter, perhaps even without the examples.

I do like having the documentation in the Dockerfiles, because then you can link to the files directly and have everything explained there.

**Comment by revett** [comment]
@metachris Is it worth adding the neo-python Dockerfile to the Docker Compose stack?

**Comment by metachris** [comment]
Sorry, not sure exactly what you mean -- which Docker Compose stack // what's the question in particular?

**Comment by metachris** [comment]
Actually, not sure the Docker compose file belongs here.

**Comment by localhuman** [comment]
@metachris you think this is good to go?

**Comment by metachris** [comment]
I think so. We can delete the docker compose file later anytime too.

**Pull request merged**